blueprints/data-solutions/vertex-mlops/main.tf

```diff
@@ -0,0 +1,267 @@
+  config_source_subnets = "LIST_OF_SUBNETWORKS"
+  subnetworks = [
+    {
+      self_link            = module.vpc-local[0].subnets["europe-west1/default"].self_link
+      config_source_ranges = ["ALL_IP_RANGES"]
+      secondary_ranges     = null
+    },
+  ]
+}
+
+
+module "project" {
+  source          = "../../../modules/project"
+  billing_account = var.billing_account_id
+  name            = var.project_id
+  prefix          = var.prefix
+  group_iam       = local.project_groups_iam
+  iam = {
+    "roles/aiplatform.user"         = [module.service-account-mlops.iam_email]
+    "roles/artifactregistry.reader" = [module.service-account-mlops.iam_email]
+    "roles/artifactregistry.writer" = [module.service-account-github.iam_email]
+    "roles/bigquery.dataEditor"     = [module.service-account-mlops.iam_email]
+    "roles/bigquery.jobUser"        = [module.service-account-mlops.iam_email]
+    "roles/bigquery.user"           = [module.service-account-mlops.iam_email]
+    "roles/cloudbuild.builds.editor" = [
+      module.service-account-mlops.iam_email,
+      module.service-account-github.iam_email
+    ]
+
+    "roles/cloudfunctions.invoker" = [module.service-account-mlops.iam_email]
+    "roles/dataflow.developer"     = [module.service-account-mlops.iam_email]
+    "roles/dataflow.worker"        = [module.service-account-mlops.iam_email]
+    "roles/iam.serviceAccountUser" = [
+      module.service-account-mlops.iam_email,
+      "serviceAccount:${module.project.service_accounts.robots.cloudbuild}"
+    ]
+    "roles/monitoring.metricWriter" = [module.service-account-mlops.iam_email]
+    "roles/run.invoker"             = [module.service-account-mlops.iam_email]
+    "roles/serviceusage.serviceUsageConsumer" = [
+      module.service-account-mlops.iam_email,
+      module.service-account-github.iam_email
+    ]
+    "roles/storage.admin" = [
+      module.service-account-mlops.iam_email,
+      module.service-account-github.iam_email
+    ]
+  }
+  labels                     = var.labels
+  org_policies               = try(var.org_policies, {})
+  parent                     = var.folder_id
+  project_create             = var.project_create
+  service_encryption_key_ids = var.kms_service_agents
+  services                   = var.project_services
+  shared_vpc_service_config = var.vpc == null ? null : {
+    host_project = local.vpc.host_project
+    # these are non-authoritative
+    service_identity_iam = {
+      "roles/compute.networkUser" = compact([
+        local.vpc_gke_service_agent ? "container-engine" : null,
+        local.vpc_cloudservices ? "cloudservices" : null
+      ])
+      "roles/compute.securityAdmin" = compact([
+        local.vpc_gke_security_admin ? "container-engine" : null,
+      ])
+      "roles/container.hostServiceAgentUser" = compact([
+        local.vpc_gke_service_agent ? "container-engine" : null
+      ])
+    }
+  }
+
+}
+
+module "service-account-mlops" {
```

Comment:
I would add the prefix.

Reply:
I don't think it is necessary. SAs need to be unique per project (not global id such as project_id or bucket name)